src/Setup/WixBA/RootView.xaml

```diff
@@ -1,125 +1,487 @@
-            <Image Grid.Row="2" Grid.Column="2" Source="resources\exit.png" DataContext="{Binding InstallationViewModel}" Visibility="{Binding ExitEnabled, Converter={StaticResource BooleanToVisibilityConverter}}" />
-
-            <!-- Cancel -->
-            <Button Grid.Row="2" Grid.Column="2" Background="#FFCB0000" DataContext="{Binding}" Visibility="{Binding CancelEnabled, Converter={StaticResource BooleanToVisibilityConverter}}" Command="{Binding CancelCommand}">Cancel</Button>
-            <TextBlock Grid.Row="2" Grid.Column="2" HorizontalAlignment="Center" VerticalAlignment="Center" FontSize="60" Foreground="White" IsHitTestVisible="False" DataContext="{Binding}"  Visibility="{Binding CancelEnabled, Converter={StaticResource BooleanToVisibilityConverter}}">X</TextBlock>
-
-        </Grid>
-
-    </Grid>
+    <i:Interaction.Behaviors>
+        <ux:WinNoSnapBehavior />
+    </i:Interaction.Behaviors>
+
+    <Border BorderBrush="{DynamicResource {x:Static SystemColors.ActiveBorderBrushKey}}" BorderThickness="0">
+        <!--  Catch MouseLeftButtonDown for Window dragging.  -->
+        <Border HorizontalAlignment="Stretch"
+                VerticalAlignment="Stretch"
+                Background="{DynamicResource {x:Static SystemColors.WindowBrushKey}}"
+                MouseLeftButtonDown="Background_MouseLeftButtonDown">
+
+            <!--  If click on no-input control (i.e. TextBlock, ProgressBar) catch click here to avoid InvalidOperationException.  -->
+            <DockPanel x:Name="AppArtBoardArea"
+                       VerticalAlignment="Stretch"
+                       MouseDown="RootView_IgnoreClick"
+                       MouseLeftButtonDown="Background_MouseLeftButtonDown"
+                       MouseUp="RootView_IgnoreClick">
+
+                <!--  Update checking marquee, Initial Actions, and Final Action Buttons, at the bottom of the layout.  -->
+                <DockPanel x:Name="ActionArea"
+                           Margin="0,0,10,10"
+                           HorizontalAlignment="Stretch"
+                           DockPanel.Dock="Bottom">
+                    <Grid x:Name="ActionGrid" HorizontalAlignment="Stretch">
+                        <Grid.RowDefinitions>
+                            <RowDefinition Height="auto" />
+                        </Grid.RowDefinitions>
+                        <Grid.ColumnDefinitions>
+                            <ColumnDefinition Width="5" />
+                            <ColumnDefinition Width="auto" />
+                            <ColumnDefinition Width="*" MinWidth="30" />
+                            <ColumnDefinition Width="auto" />
+                            <ColumnDefinition Width="5" />
+                        </Grid.ColumnDefinitions>
+
+                        <StackPanel x:Name="LeftMarqueeDcIsland"
```

Comment:
In general I find it useful to define the x:Name for most, or at least
major sections of XAML code, because when there are issues VS debugger uses
the x:Name in the Output messages.  The WixBA uses a WPF concept of Data
Context Islands (or DC Islands) where the DataContext property is set for a
group of controls, so that different behavior of a View is initialized with
possibly a different VM context than another set of controls.   When I was
first learning WPF and studying the WixBA I found the WixBA really
confusing because all the WPF purists were preaching having a single View
and ViewModel matched.  So when I understood the concept that the WixBA was
using of creating DataContext islands I found it useful to define x:Name to
make that intent clearer.  I also find it more useful to define x:Name
rather than using a comment, because comments cannot be nested, which gets
in the way of commenting out large blocks of XAML code to isolate errant
code.  (In my own projects I have implemented IoC and
PresentationTraceDebug handlers to catch these kinds of issues, but that
seemed like a more radical change than was proposed at this time.)

On Sun, Jul 31, 2016 at 11:54 PM, Sean Hall <notifications@github.com>
wrote:

> In src/Setup/WixBA/RootView.xaml
> <https://github.com/wixtoolset/wix4/pull/208#discussion_r72924043>:
>
> > +                           Margin="0,0,10,10"
> > +                           HorizontalAlignment="Stretch"
> > +                           DockPanel.Dock="Bottom">
> > +                    <Grid x:Name="ActionGrid" HorizontalAlignment="Stretch">
> > +                        <Grid.RowDefinitions>
> > +                            <RowDefinition Height="auto" />
> > +                        </Grid.RowDefinitions>
> > +                        <Grid.ColumnDefinitions>
> > +                            <ColumnDefinition Width="5" />
> > +                            <ColumnDefinition Width="auto" />
> > +                            <ColumnDefinition Width="*" MinWidth="30" />
> > +                            <ColumnDefinition Width="auto" />
> > +                            <ColumnDefinition Width="5" />
> > +                        </Grid.ColumnDefinitions>
> > +
> > +                        <StackPanel x:Name="LeftMarqueeDcIsland"
>
> I don't understand what the DcIsland suffix means.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/wixtoolset/wix4/pull/208/files/28721ab8e4985a5641b3efd7975a67cecd8f8d4a#r72924043>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHyCLQ0IYGS2OyHsgwUSS8fWCGTG42jmks5qbXwOgaJpZM4JVm8P>
> .
>